lang/java/reef-io/src/main/java/org/apache/reef/io/network/group/impl/task/OperatorTopologyStructImpl.java

```diff
@@ -234,6 +277,24 @@ private void sendToNode(final byte[] data, final ReefNetworkGroupCommProtos.Grou
         node, remove}));
     return retVal;
   }
+  
+  private byte[][] receiveArrayFromNode(final NodeStruct node, final boolean remove) {
```

Comment:
It is true that having a `receiveFromNode` and `receiveArrayFromNode` is a bit awkward (same for the other array methods). However, the biggest reason I added them is because when a `GroupCommunicationMessage` is sent, the data is sent actually as a 2d array ([GroupCommunicationMessage.java](https://github.com/apache/incubator-reef/blob/master/lang/java/reef-io/src/main/java/org/apache/reef/io/network/group/impl/GroupCommunicationMessage.java)). Hence, I though it would be silly to convert a 2d array message to a single array using the existing methods, and then converting them back to 2d form before actually sending them.